File: algorithm/src/main/java/baekjoon/p02576/Main.java
package baekjoon.p02576;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] inputs = new int[7];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextInt();
        }

        for (int ret : solve(inputs)) {
            System.out.println(ret);
        }
    }

    private static int[] solve(int[] inputs) {
        List<Integer> odds = odds(inputs);

        if (odds.size() == 0) {
            return new int[]{-1};
        }

        Collections.sort(odds);

        int sum = sum(odds);
        int min = odds.get(0);

        return new int[]{sum, min};
    }

    private static List<Integer> odds(int[] inputs) {
        List<Integer> ret = new ArrayList<>();

        for (int input : inputs) {
            if (input % 2 == 1) {
                ret.add(input);
            }
        }

        return ret;
    }

    private static int sum(List<Integer> inputs) {
        int ret = 0;

        for (int input : inputs) {
            ret += input;
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p02576/Main.kt
package baekjoon.p02576

fun main(args: Array<String>) {
    val inputs = IntArray(7)
    for (i in 0 until inputs.size) {
        inputs[i] = readLine()!!.toInt()
    }

    for (ret in solve(inputs)) {
        println(ret)
    }
}

private fun solve(inputs: IntArray): IntArray {
    val odds = inputs.filter { it % 2 == 1 }

    if (odds.isEmpty()) {
        return intArrayOf(-1)
    }

    val sum = odds.sum()
    val min = odds.min()!!.toInt()

    return intArrayOf(sum, min)
}